x-pack/filebeat/docs/inputs/input-azure-eventhub.asciidoc

```diff
@@ -50,14 +52,17 @@ A Blob Storage account is required in order to store/retrieve/update the offset
 
 ==== `storage_account`
 
-The name of the storage account.
+The name of the storage account. Required.
 
 ==== `storage_account_key`
 
-The storage account key, this key will be used to authorize access to data in your storage account.
+The storage account key, this key will be used to authorize access to data in your storage account, option is required.
 
 ==== `storage_account_container`
 
 Optional, the name of the storage account container you would like to store the offset information in.
 
+==== `override_resource_manager_endpoint`
+
+Optional, by default we are using the azure public environment, to override, users can provide a specific resource manager endpoint in order to use a different azure environment.
```

Comment:
Agree with the comments by Blake and Kaiyan, don't consider this an override, just a normal setting. Add also something about possible values. And possibly something about hybrid clouds support.
```suggestion
==== `resource_manager_endpoint`

Optional, resource manager endpoint to use, defaults to the public environment, possible values are... . If a URL is provided, the environment is obtained from the URL, this is useful in Hybrid Cloud deployments.
```
Maybe we could add an explicit option for environment URLs, instead of accepting names and urls in the same option. And maybe use something like `environment` instead of `resource_manager_endpoint`, in case we use the same settings for features that need access to other endpoints.
```suggestion
==== `environment.name`

Optional, environment to use, defaults to the public environment, possible values are... .

==== `environment.url`

Optional, loads the environment from a URL, this is useful in the Hybrid Cloud model. If this setting is used, `environment.name` is ignored.
```

Reply:
@jsoriano , the reason I went for the `override-` is because I do not want users to necessary specify this setting, this should be only used for other private env. To look for the azure resource manager endpoint might take some effort for the users and I want to avoid that in most of the cases where the are using the public cloud.
One workaround I would go for is to rename this option to `resource_manager_endpoint` but not add it in the config files so users will not be tempted to enter it if unnecessary.
Not sure what you mean by " instead of accepting names and urls in the same option" , we only accept azure resource manager endpoints, the azure-eventhub sdk can retrieve the specific environment based on that.
Also adding `environment.name` and `environment.url` might cause more confusion as users will have a hard time figuring those out, also there is no `environment.url` , we can talk about `management_portal_URL` or `resource_manager_endpoint` or `service_endpoint` but with the `resource_manager_endpoint` the SDK can help us identify the right env.

Reply:
> @jsoriano , the reason I went for the override- is because I do not want users to necessary specify this setting, this should be only used for other private env.

Other optional settings behave the same way, and don't have an `override` prefix, would it be different to other optional settings?

> To look for the azure resource manager endpoint might take some effort for the users and I want to avoid that in most of the cases where the are using the public cloud. One workaround I would go for is to rename this option to `resource_manager_endpoint` but not add it in the config files so users will not be tempted to enter it if unnecessary.

Yes, I find perfectly fine to have this option in the reference docs, but not in the config files.

> Not sure what you mean by " instead of accepting names and urls in the same option" , we only accept azure resource manager endpoints, the azure-eventhub sdk can retrieve the specific environment based on that.

Oh, ok, my mistake then, I saw that we accept URLs, and I thought we also accepted some kind of identifier there. One more reason to add some example values in the docs :slightly_smiling_face:
I have still a question about that, the URLs accepted by `EnvironmentFromURL`, are of the same kind as the ones of the resource manager endpoints?

> Also adding `environment.name` and `environment.url` might cause more confusion as users will have a hard time figuring those out, also there is no `environment.url` , we can talk about `management_portal_URL` or `resource_manager_endpoint` or `service_endpoint` but with the `resource_manager_endpoint` the SDK can help us identify the right env.

I was suggesting that to make configuration and its documentation easier. A user wanting to use the US Gov cloud would use `environment.name: usgov`, or something similar, accepted values could be in the documentation, so they don't need to look for specific endpoints in other places. A user that is using a hybrid cloud would use `environment.url: http://....` with an URL that can be accepted by `EnvironmentFromURL`.
Something like this would be specially helpful if we need to use at some point other endpoints, e.g: the Service Manager Endpoint, so we don't have to add a setting like `service_manager_endpoint`.

But I don't know much about these endpoints and URLs, so as you prefer :slightly_smiling_face:

Reply:
I answer myself after reading `EnvironmentFromURL` source. It seems that the URL accepted by this method is actually the resource manager endpoints and from that it obtains the other endpoints, so this solves my concerns about having different kind of values for hybrid clouds.

So I think that the only pending things to consider would be:
* Rename of the setting to `resource_manager_endpoint`.
* Document possible values for this setting, and how it works for hybrid clouds.